src/node_osrm_support.hpp

```diff
@@ -733,6 +733,74 @@ argumentsToTripParameter(const Nan::FunctionCallbackInfo<v8::Value> &args,
         return trip_parameters_ptr();
     }
 
+    if (obj->Has(Nan::New("roundtrip").ToLocalChecked()))
+    {
+        auto roundtrip = obj->Get(Nan::New("roundtrip").ToLocalChecked());
+        if (roundtrip->IsBoolean())
+        {
+            params->roundtrip = roundtrip->BooleanValue();
+        }
+        else
+        {
+            Nan::ThrowError("'roundtrip' param must be a boolean");
+            return trip_parameters_ptr();
+        }
+    }
+
+    if (obj->Has(Nan::New("source").ToLocalChecked()))
+    {
+        v8::Local<v8::Value> source = obj->Get(Nan::New("source").ToLocalChecked());
+
+        if (!source->IsString())
+        {
+            Nan::ThrowError("Source must be a string: [any, first]");
+            return trip_parameters_ptr();
+        }
+
+        std::string source_str = *v8::String::Utf8Value(source);
```

Comment:
Can you check for the ptr not being a nullptr?

Also please use Nan's string functions which are portable:
https://github.com/nodejs/nan/blob/master/doc/v8_misc.md

```c++
const Nan::Utf8String utf8str(source);
std::string s{*utf8str, *utf8str + utfstr.length()};